test/basicfuns.jl

```diff
@@ -44,6 +44,34 @@
     @test iszero(xlog1py(0 + im * 0, -1 + im * Inf))
 end
 
+@testset "xexpx" begin
+    for z in (0, 0.0, 0f0, false, -Inf)
+        @test iszero(@inferred xexpx(z))
+    end
+    for nan in (NaN16, NaN32, NaN64)
+        @test isnan(@inferred xexpx(nan))
+    end
+    for inf in (Inf16, Inf32, Inf64)
+        @test inf === @inferred xexpx(inf)
```

Comment:
IMO it's a bit more natural to flip LHS and RHS, i.e. use
```suggestion
        @test @inferred(xexpx(inf)) === inf
```
Not a big deal though :slightly_smiling_face: